### PR TITLE
feat(component): Add drag and drop support to table component

### PR DESCRIPTION
**Pull request by animesh1987**
## What
Add drag and drop support to `Table` component.

## Why
[Feature request for Drag and Drop](https://github.com/bigcommerce/big-design/issues/426)

## Notes
- This is an early PR to get feedback on API and code design.

## Todo 
- Add tests
- Add drag and drop to stateful table

Animated GIF-downsized (9)

ping @bigcommerce/frontend @deini @chanceaclark

### CODE REVIEW

**Comment by animesh1987** [comment]
On exploring further realised that `React.TableHTMLAttributes` already has a `draggable` and `onDragEnd` defined. Hence went with `dragAndDrop`, which is an object that has the method.

Particularly not happy with this approach, and would rather support the native type, but in that case onDragEnd interface becomes a bit annoying for end consumer in my opinion.

Happy to hear thoughts on this?

**Comment by bookernath** [comment]
FYI @bigcommerce/artemis-dt

File: packages/big-design/src/components/Table/Table.tsx [hunk 1]
```diff
@@ -118,26 +141,62 @@ const InternalTable = <T extends TableItem>(props: TableProps<T>): React.ReactEl
...
-            isSelected={isSelected}
-            item={item}
-            key={key}
-            onItemSelect={onItemSelect}
-          />
-        );
-      })}
-    </Body>
+  const renderDroppableItems = () => (
+    <Droppable droppableId="bd-droppable">
```
[the remark this answers is not in the document]
Comment (animesh1987) [reply packages/big-design/src/components/Table/Table.tsx:145]: Good point. Does it make sense to preprend this id with `tableIdRef` ?

[the remark this answers is not in the document]
Comment (deini) [reply packages/big-design/src/components/Table/Table.tsx:145]: You can use our `useUniqueId` hook.

File: packages/big-design/src/components/Table/spec.tsx [hunk 2]
```diff
@@ -469,3 +469,33 @@ describe('sortable', () => {
...
+      { header: 'Stock', hash: 'stock', render: ({ stock }: any) => stock },
+    ];
+  });
+
+  test('renders drag and drop icon', () => {
+    const { container } = render(<Table columns={columns} items={items} onRowDrop={onRowDrop} />);
+    const dragIcons = container.querySelectorAll('svg');
+
+    expect(dragIcons?.length).toBe(items.length);
+    expect(container.firstChild).toMatchSnapshot();
```
[the remark this answers is not in the document]
Comment (animesh1987) [reply packages/big-design/src/components/Table/spec.tsx]: Done

**Review by deini — APPROVED** [review]

**Pull request merged**